Library Features | Add Book, Delete Book

pointerish opened the pull request:
This PR implements the following:

- All feature requests specified here: https://www.theodinproject.com/courses/javascript/lessons/library

BrittanyBlake reviewed (changes requested):
### Status: Changes Required ♻️ 
****
### Hi Team! 👋 
You have done a wonderful job with your project so far! It looks really good! 👏 👏 👏  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- [ ] There is a bug when adding a book. When I click on the _mark as red_ toggle when adding a new book, it does not mark the book as read. I have recorded a gif for you to see 👇🏾 
![Screen Recording 2021-03-23 at 01 07 47 AM](https://user-images.githubusercontent.com/48804258/112073543-a5428780-8b74-11eb-9c28-64f4965934aa.gif)


****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

pointerish commented:
Hi TSEs,

Thanks for the feedback. We have corrected the specified bug and corrected an unnoticed one that was preventing the user to update books that were not added as the last entry by using an object to store the books instead of an array.

We also modernized the whole thing by fixing unnamed functions and using `const` instead of `let` when required.

Regards
Josias Alvarado

arslanbisharat reviewed (approved):
# Approved :clap:

Hello Developer, :wave:

Thank You for submitting this Pull Request. I am thrilled to announce that your project has met all the minimum requirements and now it is approved. :clap:

I am wishing you very good luck with the next project. :smile:

Once again, Congratulations!:joy:

**Best,
Arslan Bisharat(PT-TSE)
Slack: @arslanbisharat**
